### PR TITLE
Adding Genkit Web JS example

### PR DESCRIPTION
**Pull request by i2amsam**
Add an example using Gemini, Genkit and a lightweight webapp

### CODE REVIEW

File: gemini/js-web-genkit/.idx/dev.nix [hunk 1]
```diff
@@ -0,0 +1,39 @@
+# To learn more about how to use Nix to configure your environment
+# see: https://developers.google.com/idx/guides/customize-idx-env
+{ pkgs, ... }: {
+  channel = "stable-24.05"; # "stable-23.11" or "unstable"
```
Comment (rodydavis) [inline gemini/js-web-genkit/.idx/dev.nix:4]: You can remove the stable-23.11 here!

File: gemini/js-web-genkit/README.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
+# Genkit
+
+<a href="https://idx.google.com/new?template=https%3A%2F%2Fgithub.com%2Fi2amsam%2Fgenkit-testing">
```
Comment (rodydavis) [inline gemini/js-web-genkit/README.md:3]: Can you update this url?

File: gemini/js-web-genkit/idx-template.json [hunk 3]
```diff
@@ -0,0 +1,8 @@
+{
+  "name": "Genkit",
+  "description": "A sample app for showing using Gemini with Genkit",
+  "categories": ["Web app", "Backend"],
```
Comment (rodydavis) [inline gemini/js-web-genkit/idx-template.json:4]: Also needs to include the AI ML tag like the other Gemini template

File: gemini/js-web-genkit/package.json [hunk 4]
```diff
@@ -0,0 +1,25 @@
+{
+    "name": "genkit-gemin-web-example",
+    "version": "1.0.0",
+    "description": "This is a simple demonstration web app using the [Firebase Genkit SDK](https://github.com/firebase/genkit) to generate recipes for images.",
```
Comment (rodydavis) [inline gemini/js-web-genkit/package.json:4]: Do markdown links work here?

File: gemini/js-web-genkit/server.ts [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+
+const port = process.env.PORT || 3000;
+
+export const ai = genkit({
+    plugins: [googleAI()],
+});
+
+logger.setLogLevel('debug')
+const recipiePrompt = ai.definePrompt({
+        name: 'Recipies prompt',
```
Comment (rodydavis) [inline gemini/js-web-genkit/server.ts:15]: Maybe in the future there could be an option to use dotprompt files, or even just on the main genkit template

File: gemini/js-web-genkit/package.json [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+    "main": "index.js",
+    "scripts": {
+        "dev": "GENKIT_ENV=dev npx tsx --watch server.ts"
+    },
+    "dependencies": {
+        "@genkit-ai/googleai": "^0.9.0-rc || ^0.9",
+        "genkit": " ^0.9.0-rc || ^0.9",
+        "base64-js": "^1.5.1",
+        "express": "^4.17.13",
+        "markdown-it": "^13.0.2"
```
Comment (rodydavis) [inline gemini/js-web-genkit/package.json:14]: This can be removed

File: gemini/js-web-genkit/package.json [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+    "version": "1.0.0",
+    "description": "This is a simple demonstration web app using the [Firebase Genkit SDK](https://github.com/firebase/genkit) to generate recipes for images.",
+    "main": "index.js",
+    "scripts": {
+        "dev": "GENKIT_ENV=dev npx tsx --watch server.ts"
+    },
+    "dependencies": {
+        "@genkit-ai/googleai": "^0.9.0-rc || ^0.9",
+        "genkit": " ^0.9.0-rc || ^0.9",
+        "base64-js": "^1.5.1",
```
Comment (rodydavis) [inline gemini/js-web-genkit/package.json:12]: This can be removed

**Review by rodydavis — CHANGES_REQUESTED** [review]

File: gemini/js-web-genkit/package.json (hunk 7 above)
Comment (i2amsam) [reply gemini/js-web-genkit/package.json:12]: which one can be removed?

File: gemini/js-web-genkit/package.json (hunk 4 above)
Comment (i2amsam) [reply gemini/js-web-genkit/package.json:4]: probably not, that was auto generated

File: gemini/js-web-genkit/server.ts (hunk 5 above)
Comment (i2amsam) [reply gemini/js-web-genkit/server.ts:15]: Yeah, that's a good idea for a v2.

File: gemini/js-web-genkit/package.json (hunk 7 above)
Comment (rodydavis) [reply gemini/js-web-genkit/package.json:12]: Both base 64 and markdown it! It uses the can for markdown and base 64 via node api

File: gemini/js-web-genkit/idx-template.json (hunk 3 above)
Comment (i2amsam) [reply gemini/js-web-genkit/idx-template.json]: good call, updated

File: gemini/js-web-genkit/README.md (hunk 2 above)
Comment (i2amsam) [reply gemini/js-web-genkit/README.md]: updated

File: gemini/js-web-genkit/.idx/dev.nix (hunk 1 above)
Comment (i2amsam) [reply gemini/js-web-genkit/.idx/dev.nix]: done!

File: gemini/js-web-genkit/package.json (hunk 7 above)
Comment (i2amsam) [reply gemini/js-web-genkit/package.json]: Ah got it, updated!

**Review by rodydavis — APPROVED** [review]
LGTM

**Pull request merged**